plutus-pab/src/Cardano/Node/Mock.hs

```diff
@@ -137,26 +111,27 @@ runChainEffects ::
  -> Client.ClientHandler
  -> MVar AppState
  -> Eff (NodeServerEffects IO) a
- -> IO ([LogMessage ChainEvent], a)
+ -> IO ([LogMessage MockServerLogMsg], a)
 runChainEffects serverHandler clientHandler stateVar eff = do
     oldAppState <- liftIO $ takeMVar stateVar
     ((a, events), newState) <- liftIO
         $ runM
         $ runStderrLog
         $ interpret renderLogMessages
-        $ interpret (mapLog NodeMockNodeMsg)
         $ Eff.runState oldAppState
         $ Eff.runReader serverHandler
         $ Eff.runReader clientHandler
         $ Eff.runWriter
-        $ reinterpret (handleLogWriter @ChainEvent @[LogMessage ChainEvent] (unto return))
-        $ interpret (handleZoomedState T.chainState)
-        $ interpret (handleZoomedState T.followerState)
-        $ CE.handleChain
-        $ interpret Chain.handleControlChain
-        $ interpret (mapLog NodeServerFollowerMsg)
+        $ reinterpret (handleLogWriter @MockServerLogMsg @[LogMessage MockServerLogMsg] (unto return))
+        $ interpret (handleZoomedState chainState)
+        $ interpret (handleZoomedState followerState)
+        $ interpret (mapLog ProcessingChainEvent)
+        $ reinterpret CE.handleChain
+        $ interpret (mapLog ProcessingChainEvent)
+        $ reinterpret Chain.handleControlChain
+        $ interpret (mapLog FollowerMsg)
         $ FE.handleNodeFollower
-        $ interpret (mapLog NodeGenRandomTxMsg)
+        $ subsume
```

Comment:
@j-mueller I *think* `subsume` should be right here? type tetris first led me to something like

```
$ interpret $ (mapLog id)
```
which obviously seemed rather silly :joy:

Reply:
Yes `subsume` is right 👍